code/modules/mob/living/simple_animal/abnormality/aleph/army_in_black.dm

```diff
@@ -0,0 +1,498 @@
+	datum_reference.qliphoth_change(-1)
+	return
+
+//death counter from MoSB, it works exactly the same.
+/mob/living/simple_animal/hostile/abnormality/army/proc/OnMobDeath(datum/source, mob/living/died, gibbed)
+	SIGNAL_HANDLER
+	if(!IsContained())//Stops if it's breaching
+		return FALSE
+	if(died.z != z)//check for z-level congruency
+		return FALSE
+	if(istype(died, /mob/living/simple_animal/hostile/abnormality))//counter restored if another abnormality dies
+		datum_reference.qliphoth_change(1)
+		return TRUE
+	if(!died.mind)//if it's not a player
+		return FALSE
+	if(!ishuman(died))//all simple mobs, ordeals, other npcs except for humans
+		return FALSE
+	death_counter += 1
+	if(death_counter >= 2)
+		death_counter = 0
+		datum_reference.qliphoth_change(-1)
+	return TRUE
+
+/mob/living/simple_animal/hostile/abnormality/army/proc/FriendlyBreach(mob/living/carbon/human/user)
+	var/turf/T = get_turf (src)
+	var/mob/living/simple_animal/hostile/army_friend/A = new(get_turf(T))
+	friend_army += A
+	A.protection_target = user
+	A.GiveTarget(user)
+
+//*--Combat Mechanics--*
+
+/mob/living/simple_animal/hostile/abnormality/army/BreachEffect(mob/living/carbon/human/user)
+	..()
+	Blackify()
+	SpawnAdds()//set its alpha to 0 and make it non-dense
+	for(var/mob/living/L in protected_targets)
+		L.remove_status_effect(STATUS_EFFECT_PROTECTION)
+	density = FALSE
+	alpha = 0
+	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+
+/mob/living/simple_animal/hostile/abnormality/army/proc/SpawnAdds()
+	var/list/number_of_adds = list()
+	var/list/spawns = shuffle(GLOB.xeno_spawn)
+	for(var/i = LAZYLEN(number_of_adds), i < adds_max, i++)//# of iterations is equal to adds_max
+		for(var/turf/T in spawns)//this picks the first few shuffled xeno spawns. Maybe change it to a different type of loop
+			var/mob/living/simple_animal/hostile/army_enemy/E = new(get_turf(T))
+			number_of_adds += E//exists only for the loop
+			summoned_army += E//the actual army list
+			RegisterSignal(E, COMSIG_PARENT_QDELETING, .proc/ArmyDeath)
+			spawns -= T
+			break
+
+/mob/living/simple_animal/hostile/abnormality/army/proc/ArmyDeath(mob/E)//return to containment when all armies are dead
+	UnregisterSignal(E, COMSIG_PARENT_QDELETING)
+	summoned_army -= E
+	if(LAZYLEN(summoned_army) <=1)
+		death()//suppress the abnormality
+		return
+
+//convert all friendly units on breach
+/mob/living/simple_animal/hostile/abnormality/army/proc/Blackify()
+	for(var/mob/living/simple_animal/hostile/army_friend/A in friend_army)
+		QDEL_IN(A, 1 SECONDS)
+		QDEL_IN(A, rand(3) SECONDS)
+		var/mob/living/simple_animal/hostile/army_enemy/B = new(get_turf(A))
+		B.desc = "Yes.. we, the Army in Black.. blend into the human heart.. and drive away good thoughts.."
+		B.icon_state = "armyinblack"
+		B.icon_dead = "armyinblack_heart"
+		B.del_on_death = TRUE
+		friend_army -= A
+
+//friendly breach mob
+/mob/living/simple_animal/hostile/army_friend
```

Comment:
Does this even need to be a mob? It could just be an overlay behind the target and it'd serve just as well with less issue *and* less server performance being taken up.